reflect/src/main/java/dagger/reflect/ReflectiveModuleParser.java

```diff
@@ -75,16 +75,17 @@ static void parse(Class<?> moduleClass, @Nullable Object instance, Scope.Builder
             }
           }
         } else {
-          if (!Modifier.isStatic(method.getModifiers()) && instance == null) {
-            // Try to just-in-time create an instance of the module using a default constructor.
-            instance = maybeInstantiate(moduleClass);
-            if (instance == null) {
-              throw new IllegalStateException(moduleClass.getCanonicalName() + " must be set");
-            }
-          }
```

Comment:
Whole block moved inside `if (method.getAnnotation(Provides.class) != null) {`.
Instances were created for `@Module`s which didn't have any provides methods, just some instance methods (see `justAnInstanceMethod`)